context/src/main/java/io/opentelemetry/context/CurrentContextScheduledExecutorService.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.context;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+final class CurrentContextScheduledExecutorService extends CurrentContextExecutorService
+    implements ScheduledExecutorService {
+
+  private final ScheduledExecutorService delegate;
+
+  CurrentContextScheduledExecutorService(ScheduledExecutorService delegate) {
+    super(delegate);
+    this.delegate = delegate;
+  }
+
+  @Override
+  public ScheduledFuture<?> schedule(Runnable command, long delay, TimeUnit unit) {
+    return delegate.schedule(Context.current().wrap(command), delay, unit);
```

Comment:
```suggestion
    return ((ScheduledExecutorService) delegate())
        .schedule(Context.current().wrap(command), delay, unit);
```

Reply:
I think I prefer just keeping a local copy of the `delegate` rather than having to make this cast everywhere.